### DISCUSSION TITLE
Import CAD model

### DISCUSSION THREAD

**Discussion by Trophime** [MPh-py/MPh#77] [General]
Hi, is it possible to import CAD with mph provided that we have a Comsol licence for that?
It will  be great to add an example to demonstrate that feature if this can be done from python.

**Comment by john-hen** [comment]
Yes, it's possible. You can [access the full Comsol API](https://mph.readthedocs.io/en/1.1/demonstrations.html#access-the-full-comsol-api). In other words, virtually everything you can do with Java, you can do with Python. The only exceptions are: extending Java classes, and (probably) creating a GUI program using Comsol's built-in widgets. But as far as automating the workflow goes, there are no limitations.

That being said, I have neither used all of Comsol's features, nor will I ever, and documenting them would be way beyond the scope of this project. There are simply too many. Let's not forget that Comsol's [Programming Manual](https://www.comsol.com/documentation/6.0/COMSOL_ProgrammingReferenceManual.pdf) is over 1000 pages long.

For anything that isn't covered by MPh's API, please refer to the documentation section linked in the first paragraph. If it's not clear, I can amend or extend it. But ultimately, you'll have to figure out how to do this in Java, then do the same in Python. It's a straightforward process. Feel free to post your code here, that's what the discussion threads are for, so that users can learn from other users.
